src/firefly/java/edu/caltech/ipac/firefly/server/ws/WebDAVWorkspaceHandler.java

```diff
@@ -0,0 +1,59 @@
+package edu.caltech.ipac.firefly.server.ws;
+
+import edu.caltech.ipac.firefly.server.WorkspaceManager;
+import edu.caltech.ipac.util.AppProperties;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * API manager/handler for webdav client
+ * Class implementation of manager passed as property¬
+ * Created by ejoliet on 6/15/17.
+ */
+public class WebDAVWorkspaceHandler implements WorkspaceHandler {
+
+    String managerClass = AppProperties.getProperty("workspace.protocol.irsa.webdav", "edu.caltech.ipac.firefly.server.WebDAVWorkspaceManager");
+
+    /**
+     * map userKey and ws,
+     * TODO we don't want to create a new Wsmanager for every user access, do we?
+     */
+    Map<String, WorkspaceManager> wsPool;
+
+    public WebDAVWorkspaceHandler() {
+        wsPool = new HashMap<>();
+    }
+
+    public WorkspaceManager withCredentials(WsCredentials cred) {
+
+        synchronized (cred) {
```

Comment:
i've put it around synchronized in case we use it as a thread in a background stage. But as long as the object is not null, is fine, it will block.
The pool keys are the id (which is the unique key for identifying the manager. As a string you don't need to overwrite equals. 
The webdav implementation using http client rely on the id key, so unless the password/cookies are null, the workspace manager will access as anonymous with an id string. id can't be null. I was thinking that from the UI we could pass the cookie session as id or a random string. That needs to be discussed and added together with the file picker (probably need a saga for that i guess... what do you think?